fearless/Common/Services/PayoutRewardsService/PayoutValidatorsForNominatorFactory.swift

```diff
@@ -5,225 +5,156 @@ import IrohaCrypto
-            dependencies: dependecies
-        )
     }
 
-    private func createSubscanQueryOperation<T>(
-        address: String,
-        moduleName: String,
-        callName: String,
-        mapper: AnyMapper<JSON, T>,
-        reducer: AnyReducer<T, Set<AccountId>>
-    ) -> BaseOperation<Set<AccountId>> {
-        let url = subscanBaseURL
-            .appendingPathComponent(SubscanApi.extrinsics)
-
-        let params = SubscanQueryParams(address: address, moduleName: moduleName, callName: callName)
-
-        return SubscanQueryService(
-            url: url,
-            params: params,
-            mapper: mapper,
-            reducer: reducer,
-            initialResultValue: [],
-            subscanOperationFactory: subscanOperationFactory,
-            operationManager: operationManager
-        ).longrunOperation()
+    private func createResultFactory() -> AnyNetworkResultFactory<[AccountId]> {
+        AnyNetworkResultFactory<[AccountId]> { data in
+            guard
+                let resultData = try? JSONDecoder().decode(JSON.self, from: data),
+                let nodes = resultData.data?.query?.eraValidatorInfos?.nodes?.arrayValue
+            else { return [] }
+
+            let addressFactory = SS58AddressFactory()
+            let validators = nodes
+                .compactMap { SQEraValidatorInfo(from: $0) }
+                .compactMap { validatorInfo -> AccountAddress? in
+                    let contains = validatorInfo.others.contains(where: { $0.who == self.address })
+                    return contains ? validatorInfo.address : nil
+                }
+                .compactMap { accountAddress -> AccountId? in
+                    try? addressFactory.accountId(from: accountAddress)
+                }
+
+            return validators
+        }
     }
-}
 
-extension PayoutValidatorsForNominatorFactory: PayoutValidatorsFactoryProtocol {
-    func createResolutionOperation(for address: AccountAddress) -> CompoundOperationWrapper<[AccountId]> {
-        let controllersQueryWrapper = createControllersQueryWrapper(nominatorStashAddress: address)
-
-        let validatorsOperation: BaseOperation<[[AccountId]]> = OperationCombiningService<[AccountId]>(
-            operationManager: operationManager
-        ) {
-            let controllers = try controllersQueryWrapper.targetOperation.extractNoCancellableResultData()
-            let wrapper = try self.createValidatorsQueryWrapper(controllers: controllers)
-
-            return [wrapper]
-        }.longrunOperation()
-
-        validatorsOperation.addDependency(controllersQueryWrapper.targetOperation)
-
-        let mapOperation = ClosureOperation<[AccountId]> {
-            try validatorsOperation.extractNoCancellableResultData().flatMap { $0 }
+    private func requestParams(accountAddress: AccountAddress, erasRange: [EraIndex]?) -> String {
+        let eraFilter: String = {
+            guard let range = erasRange, range.count >= 2 else { return "" }
```

Comment:
I think here we have inclusive range. So, we can just say: `let fistRange = range.first, let lastRange = range.last` in the guard and then use them later without unwrapping?